src/js/stories/form-experiment/TextInputStateless.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { Component, PropTypes } from 'react';
+
+
+const DEFAULT_VALUE = 'value passed before mounting';
+
+
+export default class TextInputStatelessStory extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            value1: DEFAULT_VALUE,
+            value2: DEFAULT_VALUE,
+        };
+    }
+
+    _onChange(value) {
+        console.log('_onChange:', value);
+    }
+
+    _setValue(value) {
+        console.log('_setValue:', value);
+        this.setState({
+            value2: value,
+        });
+    }
+
+    render() {
+        return (
+            <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'flex-start', height: 300, justifyContent: 'space-between', padding: 10 }}>
+                <h1>Stateless</h1>
+
+                <span>
+                    <TextInputStateless onChange={value => this._onChange(value)} />
+                    &nbsp; updates as you type without extra code, without an initial value set
+                </span>
+
+                <span style={{ background: '#FF9999'}}>
+                    <TextInputStateless value={this.state.value1} />
+                    &nbsp; <b>will not update</b> as you type without extra code, because an initial value set
+                </span>
```

Comment:
This is only bug I found in the "stateless" component, although you just need a change handler defined that updates state in the parent component... which we'll probably do anyways. We could even log a warning in the component if no change handler is provided.